lib/chef/resource/windows_feature_powershell.rb

```diff
@@ -141,8 +141,8 @@ def to_formatted_array(x)
       action_class do
         # @return [Array] features the user has requested to install which need installation
         def features_to_install
-          # the intersection of the features to install & disabled features are what needs installing
-          @install ||= new_resource.feature_name & node["powershell_features_cache"]["disabled"]
+          # the intersection of the features to install & disabled/removed features are what needs installing
+          @install ||= new_resource.feature_name & ( node["powershell_features_cache"]["disabled"] | node["powershell_features_cache"]["removed"] )
```

Comment:
This won't work for any user that hasn't provided a source. We need to add a union of removed features against the existing list of features that can be installed when a source is available.

Reply:
Yes @tas50 Verified and updated for same. It is working as expected now for single as well as multi-feature installation